File: 1260.py
import sys
from collections import deque

def dfs(graph, start):
    visited = []
    stack = [start]

    while stack:
        node = stack.pop()

        if(node not in visited):
            visited.append(node)
            stack.extend(sorted(graph[node], reverse=True))

    return visited

def bfs(graph, start):
    visited = []
    queue = deque([start])

    while queue:
        node = queue.popleft()

        if(node not in visited):
            visited.append(node)
            queue.extend(sorted(graph[node]))

    return visited


n, m ,startPoint = map(int, sys.stdin.readline().rstrip().split())
graph = [[] for _ in range(n+1)]

for _ in range(m):
    node1, node2 = map(int, sys.stdin.readline().rstrip().split())
    graph[node1].append(node2)
    graph[node2].append(node1)

print(' '.join(map(str, dfs(graph, startPoint))))
print(' '.join(map(str, bfs(graph, startPoint))))

File: 1260.cpp
//백준 1260번 DFS와 BFS

#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> graph[1001];
bool visited[1001];

void dfs(int node) {
    visited[node] = true;
    cout << node << " ";

    for (int i = 0; i < graph[node].size(); i++) {
        int nextNode = graph[node][i];
        if (!visited[nextNode]) {
            dfs(nextNode);
        }
    }
}

void bfs(int start) {
    queue<int> q;
    visited[start] = true;
    q.push(start);

    while (!q.empty()) {
        int node = q.front();
        q.pop();
        cout << node << " ";

        for (int i = 0; i < graph[node].size(); i++) {
            int nextNode = graph[node][i];
            if (!visited[nextNode]) {
                visited[nextNode] = true;
                q.push(nextNode);
            }
        }
    }
}

int main() {
    int n, m, start;
    cin >> n >> m >> start;

    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        graph[u].push_back(v);
        graph[v].push_back(u);
    }

    for (int i = 1; i <= n; i++) {
        sort(graph[i].begin(), graph[i].end());
    }

    dfs(start);
    cout << endl;

    fill(visited, visited + 1001, false);

    bfs(start);
    cout << endl;

    return 0;
}

                   
/*
  fill()은 초기화 해주는 함수.
  안에는 차례대로 초기화 해줄 배열, 초기화할 범위, 초기화할 값으로 넣어준다.
*/
